File: 35E8FDAE-7659-41D7-8FED-B845964E8A93/0A00F756-A172-4D1D-A722-E53510EF7F22/counter_service.rs
use crate::types::{CounterConfig, CounterOutput};

pub struct CounterService {
    config: CounterConfig,
}

impl CounterService {
    pub fn new(config: CounterConfig) -> Self {
        assert!(
            config.from <= config.to,
            "\"from\" ({}) must be <= \"to\" ({})",
            config.from,
            config.to
        );
        Self { config }
    }

    pub fn count(&self) -> Vec<CounterOutput> {
        (self.config.from..=self.config.to)
            .map(|i| CounterOutput {
                value: i,
                label: format!("Count: {i}"),
            })
            .collect()
    }
}

File: 67BB1B16-274B-44D1-915F-97D91E38A3C2/counter.service.ts
import { CounterConfig, CounterOutput } from "./types";

export class CounterService {
  private config: CounterConfig;

  constructor(config: CounterConfig) {
    if (config.from > config.to) {
      throw new Error(`"from" (${config.from}) must be <= "to" (${config.to})`);
    }
    this.config = config;
  }

  *count(): Generator<CounterOutput> {
    for (let i = this.config.from; i <= this.config.to; i++) {
      yield { value: i, label: `Count: ${i}` };
    }
  }
}
